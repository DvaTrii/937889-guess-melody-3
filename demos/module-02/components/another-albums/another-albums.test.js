import React from "react";
import renderer from "react-test-renderer";
import AnotherAlbums from "./another-albums.jsx";

// Для тестов мы готовим отдельные моки.
// Не следует импортировать моки, которые
// применяются для приложения.
const albums = [
  {title: `Night Songs`, year: 1986},
  {title: `Long Cold Winter `, year: 1988},
  {title: `Heartbreak Station`, year: 1990}
];

const band = `Cinderella`;

// Вы можете запустить все тесты или только определённый
// npm run test.jest -- -t '<AnotherAlbums /> should render Cinderella'
describe(`Render another albums`, () => {

  it(`<AnotherAlbums /> should render Cinderella`, () => {
    const tree = renderer
      .create(<AnotherAlbums
        albums={albums}
        band={band}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`<AnotherAlbums /> should render null`, () => {
    const tree = renderer
    .create(<AnotherAlbums
      albums={[]}
      band={``}
    />)
    .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
