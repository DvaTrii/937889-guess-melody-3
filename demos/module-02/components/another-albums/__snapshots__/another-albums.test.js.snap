// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render another albums <AnotherAlbums /> should render Cinderella 1`] = `
Array [
  <h1>
    Cinderella
  </h1>,
  <ul>
    <li>
      Night Songs
    </li>
    <li>
      Long Cold Winter 
    </li>
    <li>
      Heartbreak Station
    </li>
  </ul>,
]
`;

exports[`Render another albums <AnotherAlbums /> should render null 1`] = `null`;
